File: packages/cli/src/utils/cursor-retrieval-env.ts
/**
 * Legacy import-path shim.
 *
 * Retrieval planning no longer reads user-global route state or process
 * environment to select an implementation. Host capability probing and
 * concrete Provider binding belong to Adapter/Middleware integration.
 */

export const ENV_UNKNOWN = "unknown" as const;
export type CursorRetrievalEnv = typeof ENV_UNKNOWN;

export interface CursorRetrievalEnvInfo {
  readonly env: CursorRetrievalEnv;
  readonly source: "neutral-compat";
  readonly detail: null;
}

export function detectCursorRetrievalEnvInfo(): CursorRetrievalEnvInfo {
  return {
    env: ENV_UNKNOWN,
    source: "neutral-compat",
    detail: null,
  };
}

export function detectCursorRetrievalEnv(): CursorRetrievalEnv {
  return ENV_UNKNOWN;
}

File: packages/cli/src/templates/trellis/scripts/common/cursor_retrieval_env.py
"""Legacy import-path shim for host-neutral retrieval planning.

V3 never reads process or user-global state to select a capability. Concrete
capability discovery belongs to the project resolver and adapters.
"""

from __future__ import annotations

from typing import Any


ENV_UNKNOWN = "unknown"
CursorRetrievalEnv = str


def detect_cursor_retrieval_env_info() -> dict[str, Any]:
    return {"env": ENV_UNKNOWN, "source": "neutral-compat", "detail": None}


def detect_cursor_retrieval_env() -> CursorRetrievalEnv:
    return ENV_UNKNOWN


def semantic_route_spec(_env: CursorRetrievalEnv) -> dict[str, object]:
    """Return a neutral requirement for legacy consumers."""
    return {
        "commands": [],
        "rationale_suffix": " Capability binding is deferred to the project resolver.",
        "platformNative": False,
        "semanticBackend": None,
    }
